internal/resource_cedar_schema.go

```diff
@@ -0,0 +1,158 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-faster/jx"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	issuer "terraform-provider-boxer/pkg/generated/api"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &cedarSchemaResource{}
+	_ resource.ResourceWithConfigure = &cedarSchemaResource{}
+)
+
+// CedarSchemaResource is a helper function to simplify the provider implementation.
+func CedarSchemaResource() resource.Resource {
+	return &cedarSchemaResource{}
+}
+
+// cedarSchemaResource is the resource implementation.
+type cedarSchemaResource struct {
+	issuerClient *issuer.Client
+}
+
+func (resource *cedarSchemaResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	client := getResourceIssuerClient(request, response)
+	resource.issuerClient = client
+}
+
+// Metadata returns the resource type name.
+func (resource *cedarSchemaResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = request.ProviderTypeName + "_cedar_schema"
+}
+
+// Schema defines the schema for the resource.
+func (resource *cedarSchemaResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the identity provider.",
+				Required:    true,
+			},
+			"data_json": schema.StringAttribute{
+				Description: "The schema data in JSON format.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (resource *cedarSchemaResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	plan, err := readCedarSchemaPlan(ctx, request.Plan, &response.Diagnostics)
+	if err != nil {
+		return
```

Comment:
why no
```
generateError(&response.Diagnostics, ..., err)
```
here?

Reply:
Because the error is recorded in the `readCedarSchemaPlan` method. I've added a comment to explain what's happening more explicitly